buildbot/env_config.py

```diff
@@ -0,0 +1,221 @@
+        "Get deps version from the VERSION section in dependency.conf"
+        options = self.conf.options(VERSIONS_SECTION)
+        filter_options_by_platform(options)
+
+        for opt in options:
+            value = self.conf.get(VERSIONS_SECTION, opt)
+
+            opt = opt.split('_')
+            opt = list(map(lambda x: x.upper(), opt))
+
+            # for 'ocl_cpu_rt_ver', 'ocl_gpu_rt_ver' options, ignore '_rt_' field
+            opt = [item for item in opt if item != 'RT']
+
+            opt = ''.join(opt)
+            opt = re.sub(r'(WINDOWS|WIN|LIN|LINUX)$', '', opt)
+
+            self.deps_versions[opt] = value
+
+
+    def get_deps_root_path(self):
+        "Get deps version from the ROOT section in dependency.conf"
+        options = self.conf.options(ROOT_SECTION)
+        filter_options_by_platform(options)
+
+        for opt in options:
+            value = self.conf.get(ROOT_SECTION, opt)
+            value = self.get_completed_root(value)
+            
+            opt = opt.split('_')
+            opt = list(map(lambda x: x.upper(), opt))
+            opt = ''.join(opt)
+            opt = re.sub(r'(WINDOWS|WIN|LIN|LINUX)$', '', opt)
+
+            self.deps_roots[opt] = value
+
+
+    def get_completed_root(self, path):
+        "Replace {ARCHIVE_ROOT} and {DEPS_ROOT}"
+        new_path = path.replace('{ARCHIVE_ROOT}', "$ARCHIVE_ROOT")
+        new_path = new_path.replace('{DEPS_ROOT}', "$RDRIVE_ROOT")
+        return new_path
+
+
+EXPORT_LINE = 'export {}="{}"\n'
+env_context = ""
+EXPORT_SHELL = "export.sh"
+
+
+def eset(name, value):
+    "Append env var to env_context"
+    global env_context
+    env_context += (EXPORT_LINE.format(name, value))
+
+
+def export_file():
+    "output export file"
+    f = open(EXPORT_SHELL, "w")
+    global env_context
+    contents = f.write(env_context)
+    f.close()
+
+
+def main():
+
+    deps_version_file = DEPENDENCY_FILE
+    vrd_file = VRD_FILE_LIN if platform.system()=='Linux' else VRD_FILE_WIN
+
+    deps = Dependency(deps_version_file)
+    env_dict, unset_vars  = deps.get_env_from_file(vrd_file)
+
+    for var in env_dict:
+        if var in os.environ and var not in unset_vars:
+            value = env_dict[var] + os.pathsep + f"${var}"
+        else:
+            value = env_dict[var]
+
+        if platform.system()=='Windows':
+            value_list = value.split(';')
+            for i in range(0, len(value_list)):
+                if re.search(':',value_list[i]):
+                    tmp = value_list[i].split(':')
+                    tmp[0] = '/%s' % tmp[0].lower()
+                    tmp[1] = tmp[1].replace('\\','/')
+                    value_list[i] = ''.join(tmp)
+            value = ':'.join(value_list)
+        eset(var, value)
+
+    export_file()
+
+    print(
+"""
+Open source testing enviroment has been written to "export.sh" successfully. Using the following 
+steps to complete the environment setup.
+1. Set the following variables correctly.
```

Comment:
I think that setting the environment in two steps is weird. I do believe that we can set up current console in one step